go-controller/pkg/metrics/master.go

```diff
@@ -353,25 +333,57 @@ func RegisterMasterMetrics(nbClient libovsdbclient.Client, sbClient libovsdbclie
 	})
 }
 
-// StartMasterMetricUpdater adds a goroutine that updates a "timestamp" value in the
-// nbdb every 30 seconds. This is so we can determine freshness of the database.
-// Also, update IPsec enabled or disable metric.
-func StartMasterMetricUpdater(stopChan <-chan struct{}, nbClient libovsdbclient.Client) {
-	startMasterMetricUpdaterOnce.Do(func() {
-		addIPSecMetricHandler(nbClient)
-		go func() {
-			ticker := time.NewTicker(30 * time.Second)
-			defer ticker.Stop()
-			for {
-				select {
-				case <-ticker.C:
-					updateE2ETimestampMetric(nbClient)
-				case <-stopChan:
-					return
+// RunTimestamp adds a goroutine that registers and updates timestamp metrics.
+// This is so we can determine 'freshness' of the components NB/SB DB and northd.
+// Function must be called once.
+func RunTimestamp(stopChan <-chan struct{}, sbClient, nbClient libovsdbclient.Client) {
+	// Metric named nb_e2e_timestamp is the UNIX timestamp this instance wrote to NB DB. Updated every 30s with the
+	// current timestamp.
+	prometheus.MustRegister(metricNbE2eTimestamp)
+
+	// Metric named sb_e2e_timestamp is the UNIX timestamp observed in SB DB. The value is read from the SB DB
+	// cache when metrics HTTP endpoint is scraped.
+	scrapeOvnSbE2eTimestamp := func() float64 {
+		sbGlobal, err := libovsdbops.FindSBGlobal(sbClient)
+		if err != nil {
+			klog.Errorf("Failed to get global options for the SB_Global table: %v", err)
+			return 0
+		}
+		if val, ok := sbGlobal.Options["e2e_timestamp"]; ok {
+			return parseMetricToFloat(MetricOvnkubeSubsystemMaster, "sb_e2e_timestamp", val)
+		}
+		return 0
+	}
+	prometheus.MustRegister(prometheus.NewGaugeFunc(
+		prometheus.GaugeOpts{
+			Namespace: MetricOvnkubeNamespace,
+			Subsystem: MetricOvnkubeSubsystemMaster,
+			Name:      "sb_e2e_timestamp",
+			Help:      "The current e2e-timestamp value as observed in the southbound database",
```

Comment:
Why isn't it defined together with "nb_e2e_timestamp" [here](https://github.com/ovn-org/ovn-kubernetes/pull/2839/commits/84eb498213e2d2674e45711192db51a73a968c25#diff-d26b0d0595f4dbf9b56b7f9eb180bee1d33734c47a544a72b8d1157c48e59148R36)